### PR TITLE
provides DynamicRequestHeaderController for dynamically filling RequestHeaderConfig struct

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>

/kind feature

**What this PR does / why we need it**: provides `DynamicRequestHeaderController` that combines `DynamicCAFromConfigMapController` and `RequestHeaderAuthRequestController`
into one controller that will dynamically fill `RequestHeaderConfig` struct

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**: builds on https://github.com/kubernetes/kubernetes/pull/90526

**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/options/authentication_dynamic_request_header.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		return nil, fmt.Errorf("unable to create RequestHeaderAuthRequest controller: %v", err)
+	}
+	return &DynamicRequestHeaderController{
+		ConfigMapCAController:              requestHeaderCAController,
+		RequestHeaderAuthRequestController: requestHeaderAuthRequestController,
+	}, nil
+}
+
+func (c *DynamicRequestHeaderController) RunOnce() error {
+	return c.ConfigMapCAController.RunOnce()
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/server/options/authentication_dynamic_request_header.go:55]: I thought I recalled a RunOnce on your new controller.  doesn't it make more sense to call here?

Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/server/options/authentication_dynamic_request_header.go]: done - https://github.com/kubernetes/kubernetes/pull/90548/commits/4229922c4966a726b8a12fd936d7d397f7fd711a

**Comment by deads2k** [comment]
/lgtm
/approve
/hold

squash the bazel in.

**Comment by ping035627** [comment]
please run ./hack/update-gofmt.s.

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+func (c *RequestHeaderAuthRequestController) ExtraHeaderPrefixes() []string {
+	return c.loadRequestHeaderFor(c.extraHeaderPrefixesKey)
+}
+
+func (c *RequestHeaderAuthRequestController) AllowedClientNames() []string {
+	return c.loadRequestHeaderFor(c.allowedClientNamesKey)
+}
+
+// Run starts RequestHeaderAuthRequestController controller and blocks until stopCh is closed.
+func (c *RequestHeaderAuthRequestController) Run(workers int, stopCh <-chan struct{}) {
```
Comment (tedyu) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:165]: workers parameter is not used.

**Pull request merged**